MetaMask/metamask-extension
# Updating icon imports to TS version in component-library folder

## Explanation
A part of the TypeScript migration work. This PR updates 
- All previously deprecated imports to the new TS version of the component and enums e.g. `Icon`, `IconName`, `IconSize`
- Updates all propTypes that used `Icon.propTypes` to `object`
- Update all docs that say ICON_NAMES object to `IconName` enum

The enums being used in this PR are here
`IconSize` https://github.com/MetaMask/metamask-extension/blob/develop/ui/components/component-library/icon/icon.types.ts#L4-L11
`IconName` https://github.com/MetaMask/metamask-extension/blob/develop/ui/components/component-library/icon/icon.types.ts#L22-L150

* Part of #18326 
* See: https://github.com/MetaMask/metamask-extension/pull/18431
* https://consensys.slack.com/archives/G1L7H42BT/p1680299226290989


## Screenshots/Screencaps

No change in icon display. Icon docs have been updated accordingly screencasts show no change in stories.

### Before

https://user-images.githubusercontent.com/8112138/229937890-2af9799c-39d2-4fda-8631-489b1ed283cb.mov

### After

https://user-images.githubusercontent.com/8112138/229937647-f5a3d5c2-abeb-4358-95e0-729a90853294.mov

## Manual Testing Steps

- Go to the storybook build of this PR 
- Go through stories that use the Icon component
- Check icon display and controls

## Pre-merge author checklist


## Pre-merge reviewer checklist


If further QA is required (e.g. new feature, complex testing steps, large refactor), add the `Extension QA Board` label.

In this case, a QA Engineer approval will be be required.
